src/js/components/ColumnChooserMenu.js

```diff
@@ -36,66 +88,90 @@ export default class ColumnChooserMenu extends React.Component<Props> {
     e.stopPropagation()
     this.props.dispatch(Columns.showAllColumns(this.props.tableColumns))
   }
 
   onColumnClick(e: Event, column: TableColumn) {
     e.stopPropagation()
     if (column.isVisible) {
-      if (this.allVisible()) {
-        this.props.dispatch(Columns.hideAllColumns(this.props.tableColumns))
-        this.props.dispatch(Columns.showColumn(this.tableId(), column))
-      } else if (this.props.tableColumns.visibleCount() === 1) {
-        this.props.dispatch(Columns.showAllColumns(this.props.tableColumns))
-      } else {
-        this.props.dispatch(Columns.hideColumn(this.tableId(), column))
-      }
+      this.props.dispatch(Columns.hideColumn(this.tableId(), column))
     } else {
       this.props.dispatch(Columns.showColumn(this.tableId(), column))
     }
   }
 
-  className() {
-    return classNames("column-chooser-menu", {
-      "all-visible": this.props.tableColumns.allVisible()
-    })
-  }
-
   render() {
+    const {columnHeadersView} = this.props
     const columns = this.props.tableColumns.getColumns()
     const count = this.props.tableColumns.visibleCount()
-    const allVisible = this.props.tableColumns.allVisible()
+
+    const template = [
+      {
+        label: "On",
+        click: () => this.props.dispatch(Viewer.setColumnHeadersView("ON"))
+      },
+      {
+        label: "Off",
+        click: () => this.props.dispatch(Viewer.setColumnHeadersView("OFF"))
+      },
+      {
+        label: "Auto",
+        click: () => this.props.dispatch(Viewer.setColumnHeadersView("AUTO"))
+      }
+    ]
+
+    const openMenu = usePopupMenu(template)
+    const onClick = () => {
+      openMenu()
+    }
+
     return (
       <CSSTransition
         classNames="slide-in-right"
         timeout={{enter: 150, exit: 150}}
         in={true}
         appear
       >
-        <div className={this.className()} onClick={(e) => e.stopPropagation()}>
+        <div
+          className="column-chooser-menu"
+          onClick={(e) => e.stopPropagation()}
+        >
           <Fieldset>Column Chooser</Fieldset>
           <hr />
           <CloseButton light onClick={this.props.onClose} />
-          {!allVisible && (
-            <div className="count" onClick={this.showAllColumns}>
-              <Label>{count}</Label>
-            </div>
-          )}
+          <div className="count" onClick={this.showAllColumns}>
+            <Label>{count}</Label>
+          </div>
           <ul>
-            <li className="show-all" onClick={this.showAllColumns}>
-              <Paragraph>Show All</Paragraph>
-            </li>
+            <ControlListItem>
+              <StyledLabel>Headers</StyledLabel>
+              <StyledViewSelect onClick={onClick}>
+                {columnHeadersView.toLowerCase()}
+                <DropdownArrow />
+              </StyledViewSelect>
+            </ControlListItem>
+            <ControlListItem>
+              <Paragraph onClick={this.selectAllColumns}>Select All</Paragraph>
```

Comment:
I'd like to deprecate this old `<Paragraph />` component. These would be "Label Small"